network-config-analyzer/NetworkConfig.py

```diff
@@ -206,129 +206,77 @@ def get_referenced_ip_blocks(self):
-            else:
-                if self.profiles:
-                    allowed_non_captured_conns = self._get_profile_conns(from_peer, to_peer, is_ingress).allowed_conns
-        elif pass_conns:
-            allowed_conns |= pass_conns & self._get_profile_conns(from_peer, to_peer, is_ingress).allowed_conns
-        allowed_conns -= ingress_denied_conns
-        allowed_non_captured_conns -= ingress_denied_conns
-        # It's enough that ingress impacts allowed_conns.
-        # We don't want to mix the denied_conns of the network policy by ingress,
-        # we want to preserve the specific network policy's denied connections for the output report.
-        return PolicyConnections(policy_captured, allowed_conns, denied_conns,
-                                 all_allowed_conns=allowed_conns | allowed_non_captured_conns)
-
-    def allowed_connections(self, from_peer, to_peer):
+    def get_allowed_labels(self):
+        if self.allowed_labels is not None:
+            return self.allowed_labels
+        self.allowed_labels = set()
+        for policy in self.policies.values():
+            self.allowed_labels |= policy.referenced_labels
+        return self.allowed_labels
+
+    # return the allowed connections considering all layers in the config
+    def allowed_connections(self, from_peer, to_peer, layer_name=None):
         """
         This is the core of the whole application - computes the set of allowed connections from one peer to another.
         In our connectivity model, this function computes the labels for the edges in our directed graph.
         :param Peer.Peer from_peer: The source peer
         :param Peer.Peer to_peer: The target peer
+        :param NetworkLayerName layer_name: The name of the layer to use, if requested to use a specific layer only
         :return: a 4-tuple with:
           - allowed_conns: all allowed connections (captured/non-captured)
           - captured_flag: flag to indicate if any of the policies captured one of the peers (src/dst)
           - allowed_captured_conns: allowed captured connections (can be used only if the captured flag is True)
           - denied_conns: connections denied by the policies (captured)
         :rtype: ConnectionSet, bool, ConnectionSet, ConnectionSet
         """
-        if isinstance(to_peer, Peer.IpBlock):
-            ingress_conns = PolicyConnections(captured=False, all_allowed_conns=ConnectionSet(True))
-        else:
-            ingress_conns = self._allowed_xgress_conns(from_peer, to_peer, True)
+        if layer_name is not None:
+            if layer_name not in self.layers:
+                self.layers.add_empty_layer(layer_name)
+            return self.layers[layer_name].allowed_connections(from_peer, to_peer)
+
+        # TODO initialize differently?
+        allowed_conns_res = None
+        allowed_captured_conns_res = None
+        captured_flag_res = None
+        denied_conns_res = None
```

Comment:
```suggestion
        allowed_conns_res = ConnectionSet(True)
        allowed_captured_conns_res = ConnectionSet()
        captured_flag_res = False
        denied_conns_res = ConnectionSet()
```
Why doesn't this work without the if in the loop?
Also, move right before the for loop.

Reply:
done